xopt/generators/bayesian/models/standard.py

```diff
@@ -234,7 +250,16 @@ def build_model(
                 f"could not be added to the model."
             )
 
-        return ModelListGP(*models)
+        full_model = ModelListGP(*models)
+
+        # if specified, use cached model hyperparameters
+        if self.use_cached_hyperparameters:
+            full_model.load_state_dict(self._hyperparameter_store)
```

Comment:
Does this work correctly if cuda setting is changed between invocations? Might need to move tensors to correct device.

Reply:
addressed in changes to update dtype and device when loading into model